### PR TITLE
feat: [CI-4842]: Adding `Other` option on `Select Git Provider` screen for Hosted Builds

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
### Summary

1. Added `Other` option for users that do not want to/do not have a git repo to clone.

image

image

2. Added `Clone Codebase` switch to turn `cloneCodebase` property on/off.

image

image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 2 Issues 
* Code issue 2 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10500)

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/Constants.ts [hunk 1]
```diff
@@ -172,59 +177,71 @@ const DOCKER_REGISTRY_CONNECTOR_REF = 'harnessImage'
...
+                  {
+                    step: {
+                      type: 'Run',
+                      name: 'Echo Welcome Message',
+                      identifier: 'Run',
+                      spec: {
+                        connectorRef: ACCOUNT_SCOPE_PREFIX.concat(DOCKER_REGISTRY_CONNECTOR_REF),
+                        image: 'alpine',
+                        shell: 'Sh',
+                        command: 'echo "Welcome to Harness CI" '
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/Constants.ts:222]: can remove extra space at the end

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx [hunk 2]
```diff
@@ -314,14 +317,27 @@ const SelectGitProviderRef = (
...
         setFieldTouched: formikRef.current.setFieldTouched,
         validatedConnector: connector,
         validatedSecret: secret
       })
     }
-  }, [formikRef.current?.values, formikRef.current?.setFieldTouched, connector, secret])
+  }, [formikRef, connector, secret, authMethod, gitProvider])
+
+  useEffect(() => {
+    if (gitProvider?.type === 'Other') {
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:335]: create an enum for git provider types

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx [hunk 3]
```diff
@@ -830,41 +858,79 @@ const SelectGitProviderRef = (
             <Form>
               <Container
                 padding={{ top: 'xxlarge', bottom: 'xxxlarge' }}
-                className={cx({ [css.borderBottom]: gitProvider })}
+                className={cx({ [css.borderBottom]: gitProvider && gitProvider.type !== 'Other' })}
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:861]: create an enum for git provider types

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx [hunk 4]
```diff
@@ -876,10 +942,12 @@ const SelectGitProviderRef = (
                   </Container>
                 ) : null}
               </Container>
-              {gitProvider ? (
+              {gitProvider && gitProvider.type !== 'Other' ? (
```
Comment (mtran7) [inline src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:945]: enum for git provider types

File: src/modules/75-ci/strings/strings.en.yaml [hunk 5]
```diff
@@ -181,6 +181,9 @@ getStartedWithCI:
   startProvisioning: Start Provisioning
   provisioningHelpText: Provision the '{{ $.ci.getStartedWithCI.hostedByHarness }}' Infrastructure (~2 mins)
   settingUpCIPipeline: Just a moment while we set up a CI pipeline for you...
+  nonGitCloneOption: Use this option if your git provider is not supported by Harness or if you have nothing to clone.
```
Comment (mtran7) [inline src/modules/75-ci/strings/strings.en.yaml:184]: Capital Git

**Review by mtran7 — APPROVED** [review]

File: src/modules/75-ci/strings/strings.en.yaml (hunk 5 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/strings/strings.en.yaml:184]: Done.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/Constants.ts (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/Constants.ts]: Done.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx [hunk 6]
```diff
@@ -830,41 +858,79 @@ const SelectGitProviderRef = (
...
+                    className={css.icons}
+                    cardClassName={css.gitProviderCard}
+                    renderItem={(item: GitProvider) => (
+                      <Layout.Vertical flex>
+                        <Icon
+                          name={item.icon}
+                          size={30}
+                          flex
+                          className={cx(
+                            { [css.githubIcon]: item.icon === 'github' },
```
[the remark this answers is not in the document]
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx]: Done.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx (hunk 4 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:946]: Done.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx (hunk 3 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:862]: Done.

File: src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx (hunk 2 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/pages/get-started-with-ci/InfraProvisioningWizard/SelectGitProvider.tsx:336]: Done.

**Review by d1wilko — APPROVED** [review]
No concerns here :)

**Pull request merged**